### ISSUE TITLE
Feature Request : ZeroPad3d

### ISSUE THREAD

**Issue by bharadwaj003** [pytorch/pytorch#3230]
I am using 2 x 2 (even filter size) window convolutions in 3D and hence require asymmetric padding to maintain 'same' input and output shapes. 
In torch.nn module, I can only see ZeroPad2d padding layer. 
Can you please add ZeroPad3d layer ? 
Also, can you suggest any other way to achieve asymmetric padding in 3D?

**Comment by soumith** [comment]
torch.nn.functional.pad can do asymmetric padding in N dimensions (including 3D).
http://pytorch.org/docs/master/nn.html#torch.nn.functional.pad

**Comment by hyperfraise** [comment]
I would also appreciate this feature. 2d padders aren't asked to use the functionnal api, why not 3d padders ?
